### DISCUSSION TITLE
Facecolor for crystalshape based on orientation

### DISCUSSION THREAD

**Discussion by TomPotocny** [mtex-toolbox/mtex#2254] [Ask Anything] [answered]
Hello, everybody,
I'd like to ask for your help. I would need the crystals to be the same color as the grains underneath them. I went more or less by the instructions (https://mtex-toolbox.github.io/CrystalShapes.html).
However, I have a problem, everything works as it should only the color of the crystals does not. Could you give me some advice? I have created a SHAPECOLOR where the color of each crystal is defined in rgb. However, I can't figure out how to assign it to them.
I am attaching the mentioned line, what should be in place of the question mark.
Thank you for any help

`plot(Qrains, 1.0*shape,'FaceColor',shapecolor(**?**,:),'FaceAlpha',0.7)`

![image](https://github.com/user-attachments/assets/2b3c667d-7729-44e5-9470-2f69fdb025cf)

**Comment by ralfHielscher** [comment]
what about
```matlab
plot(Qrains, 1.0*shape,'FaceColor',shapecolor',FaceAlpha',0.7)
```

**Reply by TomPotocny** [reply]
I tried it unfortunatelly, almost everything

```
Error using matlab.graphics.primitive.Patch/set
Error setting property 'FaceColor' of class 'Patch':
Invalid RGB triplet. Specify a three-element vector of values between 0 and 1.

Error in optiondraw (line 10)
if ~isempty(options), set(h,options{:});end

Error in crystalShape/plot (line 92)
h = optiondraw(patch('Faces',cS.F,'Vertices',V,'edgeColor','k'),varargin{:});

Error in grain2d/plot (line 106)
  h = plot(xy + scaling .* (rotate(varargin{1},grains.meanOrientation)),...
```

**Comment by zmichels** [comment]
Is shapecolor a list of RGB triplets?

**Reply by TomPotocny** [reply]
exactly
![image](https://github.com/user-attachments/assets/7d9c2284-dbb0-4834-9f41-0632a56818ff)

**Comment by TomPotocny** [comment]
If I define a row of RGB triplets (1) it works but of course they are all one color.

`plot(Qrains, 1.0*shape,'FaceColor',shapecolor(1,:),'FaceAlpha',0.7`

![image](https://github.com/user-attachments/assets/6ea82f16-d8e9-4d97-8467-70ffffb60060)

**Comment by ralfHielscher** [comment, marked as answer]
Hi Tom,

can you try with the new MTEX 6.0.0? I'm afraid this is a functionality I added only recently.

Ralf.

**Reply by TomPotocny** [reply]
Thank you for a advise but now there are another problems.
![image](https://github.com/user-attachments/assets/ccb8babd-0913-433d-b305-af60337a2413)

**Reply by ralfHielscher** [reply]
Can you check, that you have extracted the zip correctly. The files mentioned as missing are contained in the mtex-6.0.0.zip.
They should be in the folder `mtex/mex/`
You can also download them separately at https://github.com/mtex-toolbox/mtex/tree/develop/mex
Ralf.

**Reply by TomPotocny** [reply]
The problem was with the antivirus program, which deleted the necessary files directly after installation. Now everything works as it should. And especially my original problem.
Thank you very much for your help Ralf

![image](https://github.com/user-attachments/assets/99e569f8-a9db-45b9-b0a7-b5cf569b0c77)

**Reply by ralfHielscher** [reply]
Great! Small hint: you may type the strange command ` myaa` after your have finalized the plot. It will apply antialiasing and make your plot looking much nicer.

Ralf.
